### PR TITLE
Tx Tips in L2

### PR DESCRIPTION
**Pull request by rachel-bousfield**
Passes the info needed to compute tips to the tx processor

### CODE REVIEW

File: core/state_transition.go [hunk 1]
```diff
@@ -388,9 +389,9 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	} else {
 		fee := new(big.Int).SetUint64(st.gasUsed())
 		fee.Mul(fee, effectiveTip)
-		st.state.AddBalance(st.evm.Context.Coinbase, fee)
+		st.state.AddBalance(tipReceipient, fee)
 
-		// Arbitrum: record the tip if nonzero (this should never happen in L2)
+		// Arbitrum: record the tip
 		if st.evm.Config.Debug {
 			st.evm.Config.Tracer.CaptureArbitrumTransfer(st.evm, nil, &st.evm.Context.Coinbase, fee, false, "tip")
```
Comment (PlasmaPower) [inline core/state_transition.go:396]: We should probably always record this if `!DropTip()` for clarity (we also record other zero transfers)

**Review by PlasmaPower — COMMENTED** [review]
Generally LGTM, one minor tracing comment

File: core/state_transition.go (hunk 1 above)
Comment (rachel-bousfield) [reply core/state_transition.go]: Moved below so we can do this :)

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**